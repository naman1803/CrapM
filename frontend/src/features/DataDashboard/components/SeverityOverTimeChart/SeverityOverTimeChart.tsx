import React from 'react';
import { LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer, Label } from 'recharts';
import ChartToolTip from '../ChartToolTip/ChartToolTip';
import { Box, Tab, Tabs } from '@mui/material';
import SeverityDot from '../SeverityDot/SeverityDot';
import { Contaminants } from '@/types/SewageData';

export interface SeverityOverTimeProps {
    data: Contaminants;
}

interface GraphPanelProps {
    children?: React.ReactNode;
    index: number;
    value: number;
}

const GraphPanel: React.FC<GraphPanelProps> = ({ children, value, index }) => {
    return (
        <Box
            hidden={value !== index} // hide panel if not selected
            id={`contaminant-tabpanel-${index}`}
            aria-labelledby={`contaminant-tab-${index}`}
        >
            {value === index && (
                <Box>{children}</Box>
            )}
        </Box>
    );
}

const SeverityOverTimeChart: React.FC<SeverityOverTimeProps> = ({ data }) => {
    const [currentTab, setCurrentTab] = React.useState(0);

    // event var is needed for handle, ignoring
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const handleChange = (event: React.SyntheticEvent, newCurrentTab: number) => {
        setCurrentTab(newCurrentTab);
    };

    return (
        <Box>
            <Tabs value={currentTab} onChange={handleChange} aria-label="contaminants tab">
                {Object.entries(data).map(([keyName], index) => (
                    // create tab headers
                    <Tab
                        key={keyName}
                        label={keyName}
                        id={`contaminant-tab-${index}`}
                    />
                ))}
            </Tabs>
            {Object.entries(data).map(([keyName, values], index) => {
                // create tab panels
                return (
                    <GraphPanel key={keyName} value={currentTab} index={index}>
                        <ResponsiveContainer width="100%" height="100%" minHeight={"300px"}>
                            <LineChart
                                width={500}
                                height={300}
                                data={values.records}
                                margin={{
                                    top: 5,
                                    right: 30,
                                    left: 30,
                                    bottom: 5,
                                }}
                            >
                                <CartesianGrid vertical={false} horizontal={false} />
                                <XAxis
                                    dataKey="date"
                                    padding={{ left: 30, right: 30 }}
                                    domain={['dataMin', 'dataMax']}
                                />
                                <YAxis
                                    padding={{ top: 40, bottom: 40 }}
                                    dataKey="severityScore"
                                >
                                    <Label
                                        style={{
                                            textAnchor: "middle",
                                            fontSize: "130%",
                                            fill: "white",
                                        }}
                                        angle={270}
                                        value={'Severity Score'}
                                        dx={-40}
                                    />
                                </YAxis>
                                <Tooltip
                                    content={
                                        <ChartToolTip
                                            labels={{
                                                severityScore: "Severity Score",
                                                date: "Date"
                                            }}
                                        />
                                    }
                                />
                                <Line
                                    type="monotone"
                                    dataKey="severityScore"
                                    stroke="white"
                                    dot={({ cx, cy, payload }) => {
                                        const { severityLevel } = payload;
                                        return (
                                            <SeverityDot key={`${cx}_${cy}_${severityLevel}`} cx={cx} cy={cy} severityLevel={severityLevel} />
                                        );
                                    }}
                                />
                            </LineChart>
                        </ResponsiveContainer>
                    </GraphPanel>
                );
            })}
        </Box>
    );
};
export default React.memo(SeverityOverTimeChart);

