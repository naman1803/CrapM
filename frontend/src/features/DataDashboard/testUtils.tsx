jest.mock('recharts', () => ({
  __esModule: true,
  ...jest.requireActual('recharts'),
  ResponsiveContainer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>ResponsiveContainer</p>
      {children}
    </div>
  )),
}));
