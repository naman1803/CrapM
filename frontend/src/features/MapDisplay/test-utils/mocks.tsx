import {
  LakeGeoPoint,
  MapElement,
  MapElementType,
  SewageGeoPoint,
} from '../types/MapElements';

export const mockLakeElements: MapElement[] = [
  {
    id: 1,
    dataType: 'lake',
    isSelectable: true,
    elementType: MapElementType.Marker,
    dataPoint: {
      coordinates: [0, 0],
      id: 1,
      name: 'Grrrr Lake',
      additional_notes: 'This is a lake',
    } as LakeGeoPoint,
  },
  {
    id: 2,
    dataType: 'lake',
    isSelectable: true,
    elementType: MapElementType.Marker,
    dataPoint: {
      coordinates: [0, 0],
      id: 2,
      name: 'Obama Lake',
    } as LakeGeoPoint,
  },
];

export const mockSewageElements: MapElement[] = [
  {
    id: 1,
    dataType: 'sewage',
    isSelectable: true,
    elementType: MapElementType.Circle,
    dataPoint: {
      coordinates: [0, 0],
      id: 1,
      facility_owner: 'Eric',
      radius: 100,
      colour: 'Blue',
    } as SewageGeoPoint,
  },
  {
    id: 2,
    dataType: 'sewage',
    isSelectable: true,
    elementType: MapElementType.Circle,
    dataPoint: {
      coordinates: [0, 0],
      id: 2,
      facility_owner: 'Myron',
      radius: 100,
      colour: 'Red',
    } as SewageGeoPoint,
  },
];
