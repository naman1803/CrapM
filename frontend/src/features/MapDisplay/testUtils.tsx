import { LeafletEventHandlerFnMap, LatLngExpression } from 'leaflet';

jest.mock('react-leaflet', () => ({
  __esModule: true,
  ...jest.requireActual('leaflet'),
  useMap: () => ({
    flyTo: jest.fn(),
    getZoom: jest.fn().mockReturnValue(13), // Add getZoom method returning default zoom level
    on: jest.fn(), // Mock on method for event handling
    off: jest.fn(), // Mock off method for removing event listeners
  }),
  MapContainer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>Map Container</p>
      {children}
    </div>
  )),
  Marker: jest
    .fn()
    .mockImplementation(
      ({
        position,
        children,
        key,
        eventHandlers,
      }: {
        eventHandlers: LeafletEventHandlerFnMap;
        position: LatLngExpression;
        key: any;
        children: React.ReactNode;
      }) => (
        <div key={key}>
          <p>{`Marker-${position.toString()}`}</p>
          {Object.entries(eventHandlers).map(([keyE, value]) => (
            <button key={keyE} onClick={value}>
              {`button-${keyE}-${position.toString()}`}
            </button>
          ))}
          {children}
        </div>
      ),
    ),
  Popup: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>Popup</p>
      {children}
    </div>
  )),
  TileLayer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>TileLayer</p>
      {children}
    </div>
  )),
}));
