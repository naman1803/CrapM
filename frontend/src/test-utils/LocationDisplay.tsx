import { useLocation } from 'react-router-dom';
/**test component which displays the current url location.
 * use testid = location-display to get value of current location
 */
const LocationDisplay: React.FC = () => {
  const location = useLocation();

  return <div data-testid='location-display'>{location.pathname}</div>;
};
export default LocationDisplay;
