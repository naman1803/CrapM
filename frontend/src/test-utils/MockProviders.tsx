import { ThemeProvider } from '@mui/material';
import { MemoryRouter } from 'react-router-dom';
import { themeOptions } from '@app/theme';
import React from 'react';

// Wraps all providers for the app for testing
export const TestProviders = ({
  children,
  initialRoute = '/',
}: {
  children?: React.ReactNode;
  initialRoute?: string;
}) => {
  return (
    <ThemeProvider theme={themeOptions}>
      <MemoryRouter initialEntries={[initialRoute]}>{children}</MemoryRouter>
    </ThemeProvider>
  );
};
