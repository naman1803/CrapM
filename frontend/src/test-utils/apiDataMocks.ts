import { LakeData } from '@/types/LakeData';
import {
  SewageCoordData,
  SewageData,
  SewageSeverityData,
} from '@/types/SewageData';

export const mockSewageSeverityElement: SewageSeverityData = {
  id: 1,
  facilityOwner: 'Eric',
  contaminants: {
    brainrot: {
      averageSeverityScore: 1.0,
      overallSeverityLevel: 'Moderate',
      averagePercentExceedance: 100.0,
      records: [
        {
          severityScore: 1.0,
          severityLevel: 'Moderate',
          date: '2021-01-01',
        },
        {
          severityScore: 2.0,
          severityLevel: 'High',
          date: '2021-01-02',
        },
      ],
    },
    takingDumb: {
      averageSeverityScore: 2.0,
      overallSeverityLevel: 'High',
      averagePercentExceedance: 200.0,
      records: [
        {
          severityScore: 0.5,
          severityLevel: 'High',
          date: '2021-01-01',
        },
        {
          severityScore: 2.0,
          severityLevel: 'High',
          date: '2021-01-02',
        },
      ],
    },
  },
};

export const mockLakeApiData: LakeData[] = [
  {
    id: 1,
    lakeName: 'Lake Ontario',
    latitude: 45.2,
    longitude: -78.94,
    additionalNotes: null,
  },
  {
    id: 2,
    lakeName: 'Lake Erie',
    latitude: 46.47,
    longitude: -82.78,
    additionalNotes: null,
  },
];

export const mockSewageApiDataSent = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    latitude: 44.49,
    longitude: -77.69,
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    radius: 22.539892,
    colour: "Green"
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    latitude: 44.68,
    longitude: -79.23,
    contaminant: 'PH - HIGH',
    radius: 22.524761,
    colour: "Green"
  },
];
export const mockSewageApiData: SewageCoordData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    latitude: 44.49,
    longitude: -77.69,
    radius: 22.539892,
    colour: "Green"
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    latitude: 44.68,
    longitude: -79.23,
    radius: 22.524761,
    colour: "Green"
  },
];

export const mockSewageSeverityApiData: SewageSeverityData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    contaminants: {
      'BIOCHEMICAL OXYGEN DEMAND': {
        averageSeverityScore: 0.77,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 177.0,
        records: [
          {
            severityScore: 0.77,
            severityLevel: 'Low',
            date: '2016-08-18',
          },
        ],
      },
    },
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    contaminants: {
      'PH - HIGH': {
        averageSeverityScore: 0.95,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 195.4,
        records: [
          {
            severityScore: 0.95,
            severityLevel: 'Low',
            date: '2016-08-30',
          },
        ],
      },
      'BIOCHEMICAL OXYGEN DEMAND': {
        averageSeverityScore: 0.58,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 157.8,
        records: [
          {
            severityScore: 0.58,
            severityLevel: 'Low',
            date: '2016-08-23',
          },
        ],
      },
    },
  },
];

export const mockSewageAllApiData: SewageData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    siteAddress: '123 Mock St',
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-18T00:00:00',
    exceedanceEndDate: '2016-08-18T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 88.5,
    quantityMinimum: null,
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    siteAddress: '1234 Mock St',
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-23T00:00:00',
    exceedanceEndDate: '2016-08-23T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 78.9,
    quantityMinimum: null,
  },
  {
    id: 3,
    facilityOwner: 'Mock Facility 2',
    siteAddress: '1234 Mock St',
    contaminant: 'PH - HIGH',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-30T00:00:00',
    exceedanceEndDate: '2016-08-30T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 97.7,
    quantityMinimum: null,
  },
];
