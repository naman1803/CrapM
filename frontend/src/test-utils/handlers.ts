import { http, HttpResponse } from 'msw';
import {
  mockLakeApiData,
  mockSewageAllApiData,
  mockSewageApiDataSent,
  mockSewageSeverityApiData,
} from './apiDataMocks';

const handlers = [
  http.get('/api/maps/lakes/all', () => {
    return HttpResponse.json(mockLakeApiData);
  }),
  http.get('/api/maps/sewage/all', () => {
    return HttpResponse.json(mockSewageApiDataSent);
  }),
  http.get('/api/data/sewage/all', () => {
    return HttpResponse.json(mockSewageAllApiData);
  }),
  ...mockSewageSeverityApiData.map((severityData) =>
    http.get(`/api/data/sewage/severity/${severityData.id}`, () => {
      return HttpResponse.json(severityData);
    }),
  ),
];
export default handlers;
