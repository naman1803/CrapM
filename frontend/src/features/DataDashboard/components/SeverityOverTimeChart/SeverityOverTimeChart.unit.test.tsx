import './../../testUtils';

import { TestProviders } from "@/test-utils/MockProviders";
import SeverityOverTimeChart, { SeverityOverTimeProps } from "./SeverityOverTimeChart";
import { render } from "@testing-library/react";

export interface Contaminants {
    [key: string]: {
        averageSeverityScore: number;
        overallSeverityLevel: string;
        averagePercentExceedance: number;
        records: ContaminantRecord[];
    };
}

interface ContaminantRecord {
    date: string;
    severityScore: number;
    severityLevel: string;
}

const mockSeverityOverTimeData: SeverityOverTimeProps = {
    data: {
        'Metrolinx - GO Transit Halton Hills': {
            overallSeverityLevel: "Moderate",
            averagePercentExceedance: 3,
            averageSeverityScore: 2.85,
            records: [
                { severityLevel: 'Low', severityScore: 0.91, date: '2016-02-05' },
                { severityLevel: 'Moderate', severityScore: 3.22, date: '2016-02-17' },
                { severityLevel: 'Moderate', severityScore: 1.07, date: '2016-05-12' },
                { severityLevel: 'Low', severityScore: 0.67, date: '2016-05-15' },
                { severityLevel: 'Critical', severityScore: 5.01, date: '2016-06-15' },
                { severityLevel: 'High', severityScore: 3.92, date: '2016-08-02' },
                { severityLevel: 'Low', severityScore: 0.67, date: '2016-09-05' },
                { severityLevel: 'Low', severityScore: 0.57, date: '2016-10-21' },
                { severityLevel: 'Low', severityScore: 0.45, date: '2016-10-31' },
                { severityLevel: 'Low', severityScore: 0.15, date: '2016-11-07' },
            ]
        }
    }
};

describe('SeverityOverTimeChart', () => {
    it('should render the chart', () => {
        const { asFragment } = renderSeverityOverTimeChart(mockSeverityOverTimeData);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should display at least one line segment on the chart', () => {
        const { container } = renderSeverityOverTimeChart(mockSeverityOverTimeData);
        const lineTag = container.querySelector('line');
        expect(lineTag).toBeInTheDocument();
    });
});

const renderSeverityOverTimeChart = (props: SeverityOverTimeProps) => render(
    <TestProviders>
        <SeverityOverTimeChart {...props} />
    </TestProviders>
);
