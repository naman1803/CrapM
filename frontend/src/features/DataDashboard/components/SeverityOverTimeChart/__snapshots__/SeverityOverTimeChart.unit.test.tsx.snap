// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeverityOverTimeChart should render the chart 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiTabs-root css-jhczcq-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          aria-label="contaminants tab"
          class="MuiTabs-flexContainer css-1wxkzlj-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-aynvm1-MuiButtonBase-root-MuiTab-root"
            id="contaminant-tab-0"
            role="tab"
            tabindex="0"
            type="button"
          >
            Metrolinx - GO Transit Halton Hills
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-imd4kt-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      aria-labelledby="contaminant-tab-0"
      class="MuiBox-root css-0"
      id="contaminant-tabpanel-0"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div>
          <p>
            ResponsiveContainer
          </p>
          <div
            class="recharts-wrapper"
            style="position: relative; cursor: default; width: 500px; height: 300px;"
          >
            <svg
              class="recharts-surface"
              height="300"
              style="width: 100%; height: 100%;"
              viewBox="0 0 500 300"
              width="500"
            >
              <title />
              <desc />
              <defs>
                <clippath
                  id="recharts1-clip"
                >
                  <rect
                    height="260"
                    width="380"
                    x="90"
                    y="5"
                  />
                </clippath>
              </defs>
              <g
                class="recharts-cartesian-grid"
              />
              <g
                class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
              >
                <line
                  class="recharts-cartesian-axis-line"
                  fill="none"
                  height="30"
                  orientation="bottom"
                  stroke="#666"
                  width="380"
                  x="90"
                  x1="90"
                  x2="470"
                  y="265"
                  y1="265"
                  y2="265"
                />
                <g
                  class="recharts-cartesian-axis-ticks"
                >
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="120"
                      x2="120"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="120"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="120"
                      >
                        2016-02-05
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="155.55555555555554"
                      x2="155.55555555555554"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="155.55555555555554"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="155.55555555555554"
                      >
                        2016-02-17
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="191.11111111111111"
                      x2="191.11111111111111"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="191.11111111111111"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="191.11111111111111"
                      >
                        2016-05-12
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="226.66666666666669"
                      x2="226.66666666666669"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="226.66666666666669"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="226.66666666666669"
                      >
                        2016-05-15
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="262.22222222222223"
                      x2="262.22222222222223"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="262.22222222222223"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="262.22222222222223"
                      >
                        2016-06-15
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="297.77777777777777"
                      x2="297.77777777777777"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="297.77777777777777"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="297.77777777777777"
                      >
                        2016-08-02
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="333.33333333333337"
                      x2="333.33333333333337"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="333.33333333333337"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="333.33333333333337"
                      >
                        2016-09-05
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="368.8888888888889"
                      x2="368.8888888888889"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="368.8888888888889"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="368.8888888888889"
                      >
                        2016-10-21
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="404.44444444444446"
                      x2="404.44444444444446"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="404.44444444444446"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="404.44444444444446"
                      >
                        2016-10-31
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      orientation="bottom"
                      stroke="#666"
                      width="380"
                      x="90"
                      x1="440"
                      x2="440"
                      y="265"
                      y1="271"
                      y2="265"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      orientation="bottom"
                      stroke="none"
                      text-anchor="middle"
                      width="380"
                      x="440"
                      y="273"
                    >
                      <tspan
                        dy="0.71em"
                        x="440"
                      >
                        2016-11-07
                      </tspan>
                    </text>
                  </g>
                </g>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
              >
                <line
                  class="recharts-cartesian-axis-line"
                  fill="none"
                  height="260"
                  orientation="left"
                  stroke="#666"
                  width="60"
                  x="30"
                  x1="90"
                  x2="90"
                  y="5"
                  y1="5"
                  y2="265"
                />
                <g
                  class="recharts-cartesian-axis-ticks"
                >
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="260"
                      orientation="left"
                      stroke="#666"
                      width="60"
                      x="30"
                      x1="84"
                      x2="90"
                      y="5"
                      y1="225"
                      y2="225"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="260"
                      orientation="left"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="82"
                      y="225"
                    >
                      <tspan
                        dy="0.355em"
                        x="82"
                      >
                        0
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="260"
                      orientation="left"
                      stroke="#666"
                      width="60"
                      x="30"
                      x1="84"
                      x2="90"
                      y="5"
                      y1="180"
                      y2="180"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="260"
                      orientation="left"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="82"
                      y="180"
                    >
                      <tspan
                        dy="0.355em"
                        x="82"
                      >
                        2
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="260"
                      orientation="left"
                      stroke="#666"
                      width="60"
                      x="30"
                      x1="84"
                      x2="90"
                      y="5"
                      y1="135"
                      y2="135"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="260"
                      orientation="left"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="82"
                      y="135"
                    >
                      <tspan
                        dy="0.355em"
                        x="82"
                      >
                        4
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="260"
                      orientation="left"
                      stroke="#666"
                      width="60"
                      x="30"
                      x1="84"
                      x2="90"
                      y="5"
                      y1="90"
                      y2="90"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="260"
                      orientation="left"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="82"
                      y="90"
                    >
                      <tspan
                        dy="0.355em"
                        x="82"
                      >
                        6
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="260"
                      orientation="left"
                      stroke="#666"
                      width="60"
                      x="30"
                      x1="84"
                      x2="90"
                      y="5"
                      y1="45"
                      y2="45"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="260"
                      orientation="left"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="82"
                      y="45"
                    >
                      <tspan
                        dy="0.355em"
                        x="82"
                      >
                        8
                      </tspan>
                    </text>
                  </g>
                </g>
                <text
                  class="recharts-text recharts-label"
                  fill="#808080"
                  offset="5"
                  style="text-anchor: middle; font-size: 130%; fill: white;"
                  text-anchor="middle"
                  transform="rotate(270, 20, 135)"
                  x="20"
                  y="135"
                >
                  <tspan
                    dy="0.355em"
                    x="20"
                  >
                    Severity Score
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-line"
              >
                <path
                  class="recharts-curve recharts-line-curve"
                  d="M120,204.525C131.85185185185185,178.5375,143.7037037037037,152.54999999999998,155.55555555555554,152.54999999999998C167.4074074074074,152.54999999999998,179.25925925925927,194.925,191.11111111111111,200.925C202.96296296296296,206.925,214.81481481481484,209.925,226.66666666666669,209.925C238.51851851851853,209.925,250.37037037037038,112.275,262.22222222222223,112.275C274.0740740740741,112.275,285.9259259259259,120.52500000000003,297.77777777777777,136.8C309.6296296296296,153.07500000000002,321.4814814814815,208.425,333.33333333333337,209.925C345.1851851851852,211.425,357.03703703703707,211.35000000000002,368.8888888888889,212.175C380.74074074074076,213,392.5925925925926,213.3,404.44444444444446,214.875C416.2962962962963,216.45,428.14814814814815,219.0375,440,221.625"
                  fill="none"
                  height="260"
                  stroke="white"
                  stroke-dasharray="0px 0px"
                  stroke-width="1"
                  width="380"
                />
              </g>
            </svg>
            <div
              class="recharts-tooltip-wrapper"
              style="visibility: hidden; pointer-events: none; position: absolute; top: 0px; left: 0px;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
